tests/Unit/Evolution/Systems/ScalarWave/Test_Constraints.cpp

```diff
@@ -0,0 +1,351 @@
+          ::Tags::Coordinates<dim, Frame::Inertial>, ScalarWave::Psi,
+          ScalarWave::Phi<dim>, ScalarWave::Pi,
+          ::Tags::deriv<ScalarWave::Psi, tmpl::size_t<dim>, Frame::Inertial>,
+          ::Tags::deriv<ScalarWave::Phi<dim>, tmpl::size_t<dim>,
+                        Frame::Inertial>,
+          ::Tags::deriv<ScalarWave::Pi, tmpl::size_t<dim>, Frame::Inertial>>,
+      db::AddComputeTags<ScalarWave::Tags::OneIndexConstraintCompute<dim>,
+                         ScalarWave::Tags::TwoIndexConstraintCompute<dim>>>(
+      x, psi, phi, pi, deriv_psi, deriv_phi, deriv_pi);
+
+  // Compute tested quantities locally
+  const auto one_index_constraint =
+      ScalarWave::one_index_constraint(deriv_psi, phi);
+  const auto two_index_constraint = ScalarWave::two_index_constraint(deriv_phi);
+
+  // Check that their compute items in databox furnish identical values
+  CHECK(db::get<ScalarWave::Tags::OneIndexConstraint<dim>>(box) ==
+        one_index_constraint);
+  CHECK(db::get<ScalarWave::Tags::TwoIndexConstraint<dim>>(box) ==
+        two_index_constraint);
+}
+
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Evolution.Systems.ScalarWave.Constraints",
+                  "[Unit][Evolution]") {
+  pypp::SetupLocalPythonEnvironment local_python_env{
+      "Evolution/Systems/ScalarWave/"};
+  // Test the one-index constraint with random numbers
+  test_one_index_constraint_random<1>(
+      DataVector(4, std::numeric_limits<double>::signaling_NaN()));
+  test_one_index_constraint_random<2>(
+      DataVector(4, std::numeric_limits<double>::signaling_NaN()));
+  test_one_index_constraint_random<3>(
+      DataVector(4, std::numeric_limits<double>::signaling_NaN()));
+
+  // Test the two-index constraint with random numbers
+  test_two_index_constraint_random<1>(
+      DataVector(4, std::numeric_limits<double>::signaling_NaN()));
+  test_two_index_constraint_random<2>(
+      DataVector(4, std::numeric_limits<double>::signaling_NaN()));
+  test_two_index_constraint_random<3>(
+      DataVector(4, std::numeric_limits<double>::signaling_NaN()));
+
+  // Test constraints against Spherical wave
+  const size_t grid_size = 8;
+  const std::array<double, 3> upper_bound{{0.82, 1.24, 1.32}};
+  const std::array<double, 3> lower_bound{{0.8, 1.22, 1.30}};
+
+  const ScalarWave::Solutions::RegularSphericalWave solution(
+      std::make_unique<MathFunctions::Gaussian>(1., 1., 0.));
+
+  // Note: looser numerical tolerance because this check
+  // uses numerical derivatives
+  test_one_index_constraint_analytic(
+      solution, grid_size, lower_bound, upper_bound,
+      std::numeric_limits<double>::epsilon() * 1.e6);
+
+  // Note: looser numerical tolerance because this check
+  // uses numerical derivatives
+  test_two_index_constraint_analytic(
+      solution, grid_size, lower_bound, upper_bound,
+      std::numeric_limits<double>::epsilon() * 1.e6);
+}
+
+SPECTRE_TEST_CASE("Unit.Evolution.Systems.ScalarWave.ConstraintComputeTags",
```

Comment:
Can you combine the two `SPECTRE_TEST_CASE`s? We'd like to keep test cases to a minimum in order to reduce test runtime. Either scoping `{...}` with an `INFO(...);` at the beginning of each scope, or functions with `INFO(...);` at the beginning is what we've been doing. The `INFO`s are useful for giving context as to where the test failed and what it should be doing

Reply:
combined